src/checkstyle/checks.xml

```diff
@@ -32,6 +32,10 @@
         <property name="message" value="Blank line before end of file" />
     </module>
     <module name="RegexpSingleline">
+        <property name="format" value="(class|interface) ([a-zA-Z0-9])* (extends|implements)" />
```

Comment:
I think we can match generic classes like this:
```
(class|interface) ([a-zA-Z0-9])+(<.*>)? (extends|implements)
```
Note that the name part should be `+` rather than `*` since names are non-empty.